src/util/replace_symbol.cpp

```diff
@@ -89,33 +91,75 @@ bool replace_symbolt::replace(exprt &dest) const
   if(!have_to_replace(dest))
     return result;
 
-  if(dest.id()==ID_symbol)
+  if(dest.id()==ID_member)
+  {
+    member_exprt &me=to_member_expr(dest);
+
+    if(!replace(me.struct_op(), replace_with_const))
+      result=false;
+  }
+  else if(dest.id()==ID_index)
+  {
+    index_exprt &ie=to_index_expr(dest);
+
+    if(!replace(ie.array(), replace_with_const))
+      result=false;
+
+    if(!replace(ie.index()))
+      result=false;
+  }
+  else if(dest.id()==ID_address_of)
+  {
+    address_of_exprt &aoe=to_address_of_expr(dest);
+
+    if(!replace(aoe.object(), false))
+      result=false;
+  }
+  else if(dest.id()==ID_symbol)
   {
+    const symbol_exprt &s=to_symbol_expr(dest);
+
     expr_mapt::const_iterator it=
-      expr_map.find(dest.get(ID_identifier));
+      expr_map.find(s.get_identifier());
 
     if(it!=expr_map.end())
     {
-      dest=it->second;
+      const exprt &e=it->second;
+
+      if(!replace_with_const && e.is_constant())
```

Comment:
I don't think it works because you need to to_symbol_expr to get the symbol to look up in the substitution table, which gives you e (what you are substituting) and you need that for the condition (because it is only an issue if you are replacing with a constant).